ha/setup/cluster_validator/cluster_test.py

```diff
@@ -0,0 +1,371 @@
+                # identical resources
+                if desc["provider"]["counters"]:
+                    for c in desc["provider"]["counters"]:
+                        name_list.append(f"{res}-{c}")
+                else:
+                    name_list = [res]
+                for res_name in name_list:
+                    # Check that resource actually exists
+                    if desc["ha"]["mode"] == "active_active":
+                        if desc["group"] != "":
+                            r = self.status.get_resource_from_cloned_group_by_name(res_name)
+                            resource_list.append(r)
+                        else:
+                            r = self.status.get_clone_resource_by_name(res_name)
+                            resource_list.extend(r.copies)
+                    else:
+                        r = self.status.get_unique_resource_by_name(res_name)
+                        resource_list.append(r)
+
+                    if not resource_list:
+                        Log.info(f"Resource {res_name} not found in status")
+                        check_res = False
+                        continue
+            except Exception:
+                check_res = False
+                continue
+
+            for r in resource_list:
+                # Check provider and service
+                expected = "{}:{}".format(
+                    desc["provider"]["name"], desc["provider"]["service"])
+                actual = r.resource_agent
+                if expected != actual:
+                    Log.info(
+                        f"{res}: invalid resource agent is used {actual} instead of {expected}")
+                    check_res = False
+
+                if desc["group"] != r.group:
+                    Log.info(f'{res}: wrong group {r.group} vs expected {desc["group"]}')
+                    check_res = False
+                # TODO: Location to be checked once component files become part of provisioning
+
+        return check_res
+
+    def check_nodes_configured(self) -> bool:
+        """Compare expected set of nodes against existing one."""
+        expected = set(self.layout.nodes)
+        actual = set([node.name for node in self.status.get_nodes()])
+        amount_ok = len(expected) == len(actual)
+        if not amount_ok:
+            Log.info("Amount of nodes does not correspond: {} layout vs {} actual". format(
+                     len(expected), len(actual)))
+        content_ok = expected == actual
+        if not content_ok:
+            Log.info("Expected set of nodes is not equal to actual set of nodes. Diff: {}".format(
+                     expected.symmetric_difference(actual)))
+        return amount_ok and content_ok
+
+
+def _get_component_json_files(comp_files_dir: str = COMPONENTS_CONFIG_DIR) -> List[str]:
+    p = Path(comp_files_dir)
+    return list(map(str, p.glob("*.json")))
+
+
+def validate_cluster(node_list: list[str],
```

Comment:
Need to think how to differentiate VM environment from real hardware.
Since the reference layout of resources is supplied by component files, I see no problem to add boolean or integer flag there. This flag will be processed in ClusterLayout class.
validate_cluster shall also implement similar argument to pass it to ClusterLayout class in order to communicate which environment is expected.

(Note: better to do it in separate PR)

Reply:
Proposal is documented in LLD: https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/286459591/HA+Cluster+sanity+checks